maven-core/src/main/java/org/apache/maven/internal/aether/LoggingRepositoryListener.java

```diff
@@ -63,48 +63,47 @@ public void metadataResolved(RepositoryEvent event) {
     public void metadataInvalid(RepositoryEvent event) {
         Exception exception = event.getException();
 
-        StringBuilder buffer = new StringBuilder(256);
-        buffer.append("The metadata ");
+        Object metadata;
         if (event.getMetadata().getFile() != null) {
-            buffer.append(event.getMetadata().getFile());
+            metadata = event.getMetadata().getFile();
         } else {
-            buffer.append(event.getMetadata());
+            metadata = event.getMetadata();
         }
 
+        String errorType = " is invalid";
         if (exception instanceof FileNotFoundException) {
-            buffer.append(" is inaccessible");
-        } else {
-            buffer.append(" is invalid");
+            errorType = " is inaccessible";
         }
 
+        String msg = "";
         if (exception != null) {
-            buffer.append(": ");
-            buffer.append(exception.getMessage());
+            msg = ": " + exception.getMessage();
         }
 
         if (logger.isDebugEnabled()) {
-            logger.warn(buffer.toString(), exception);
+            logger.warn("The metadata {} {}{}", metadata, errorType, msg, exception);
         } else {
-            logger.warn(buffer.toString());
+            logger.warn("The metadata {} {}{}", metadata, errorType, msg);
         }
     }
 
     @Override
     public void artifactDescriptorInvalid(RepositoryEvent event) {
-        StringBuilder buffer = new StringBuilder(256);
-        buffer.append("The POM for ");
-        buffer.append(event.getArtifact());
-        buffer.append(" is invalid, transitive dependencies (if any) will not be available");
-
         if (logger.isDebugEnabled()) {
-            logger.warn(buffer + ": " + event.getException().getMessage());
+            logger.warn(
+                    "The POM for {} is invalid, transitive dependencies (if any) will not be available: {}",
+                    event.getArtifact(),
+                    event.getException().getMessage());
         } else {
-            logger.warn(buffer + ", enable verbose output (-X) for more details");
+            logger.warn(
+                    "The POM for {} is invalid, transitive dependencies (if any) will not be available, enable "
+                            + "verbose output (-X) for more details",
+                    event.getArtifact());
```

Comment:
The code above is wrong and does not make much sense. To be consistent with other parts of this class, the whole method should look like:
```
        if (logger.isDebugEnabled()) {
            logger.warn(
                "The POM for {} is invalid, transitive dependencies (if any) will not be available: {}",
                event.getArtifact(),
                event.getException().getMessage(),
                event.getException());
        } else {
            logger.warn(
                "The POM for {} is invalid, transitive dependencies (if any) will not be available: {}",
                event.getArtifact(),
                event.getException().getMessage());
        }
```